### PR TITLE
Domain Tests

### PR DESCRIPTION
**Pull request by sdsantos**
Closes #1448

We've refactored domain logic code out of Activities/Fragments/Services and into special purpose classes, so we could write unit tests for it. Besides moving the code, we tried changing it as little as possible.

---

This PR is based on the previous unmerged PR https://github.com/ooni/probe-android/pull/431
Once that one gets merged, we'll rebase this PR.

### CODE REVIEW

File: app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java [hunk 1]
```diff
@@ -66,23 +76,7 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
-                        .withNegativeButton(getString(R.string.Modal_NoThanks))
-                        .withNeutralButton(getString(R.string.Modal_DontAskAgain))
-                        .withExtra(AUTOTEST_DIALOG)
-                        .build().show(getSupportFragmentManager(), null);
-            }
-            else if (getPreferenceManager().getAppOpenCount() != 0
-                    && getPreferenceManager().getAppOpenCount() % PreferenceManager.NOTIFICATION_DIALOG_COUNT == 0
-                    && !getPreferenceManager().isNotifications()
-                    && !getPreferenceManager().isAskNotificationDialogDisabled()) {
+            if (notificationManager.shouldShow()) {
```
Comment (lorenzoPrimi) [inline app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java:79]: This seems to be incorrect, the type of modals are two, one for notifications and one for autotest

Comment (Filmaluco) [reply app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java:79]: Sorry about that, I did a fresh commit with the updated version let me know

File: app/src/main/java/org/openobservatory/ooniprobe/domain/UpdatesNotificationManager.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package org.openobservatory.ooniprobe.domain;
```
[the remark this answers is not in the document]
Comment (Filmaluco) [reply app/src/main/java/org/openobservatory/ooniprobe/domain/UpdatesNotificationManager.java:1]: Bad rebase :) will make sure only 1 stays

File: app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java [hunk 3]
```diff
@@ -66,38 +76,32 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
                 }
             });
             bottomNavigation.setSelectedItemId(getIntent().getIntExtra(RES_ITEM, R.id.dashboard));
-            if (getPreferenceManager().getAppOpenCount() != 0
-                    && getPreferenceManager().getAppOpenCount() % PreferenceManager.AUTOTEST_DIALOG_COUNT == 0
-                    && !getPreferenceManager().isAutomaticTestEnabled()
-                    && !getPreferenceManager().isAskAutomaticTestDialogDisabled()) {
+            if (notificationManager.shouldShow()) {
```
Comment (lorenzoPrimi) [inline app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java:79]: These are inverted, first ask for autorun, else notification.
see https://github.com/ooni/probe-android/blob/master/app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java#L69

File: app/src/main/java/org/openobservatory/ooniprobe/common/service/ServiceUtil.java [hunk 4]
```diff
@@ -42,72 +38,69 @@ public static void scheduleJob(Context context) {
...
+        }
     }
 
     public static void callCheckInAPI(Application app) {
+        app.getServiceComponent().inject(d);
+
         BatteryManager batteryManager = (BatteryManager) app.getSystemService(Context.BATTERY_SERVICE);
-        PreferenceManager pm = app.getPreferenceManager();
-        //Double checks for charging and wifi
-        if (pm.testWifiOnly() &&
```
Comment (lorenzoPrimi) [inline app/src/main/java/org/openobservatory/ooniprobe/common/service/ServiceUtil.java:66]: Where does this gode go? (until line 75) can't find it anywhere

Comment (Filmaluco) [reply app/src/main/java/org/openobservatory/ooniprobe/common/service/ServiceUtil.java:66]: it's inside the GenerateAutoRunServiceSuite method shouldStart. 

I will extract it to here

Comment (lorenzoPrimi) [reply app/src/main/java/org/openobservatory/ooniprobe/common/service/ServiceUtil.java:66]: Looks great

**Review by lorenzoPrimi — APPROVED** [review]

**Pull request merged**